### DISCUSSION TITLE
Capacitor 3 : Blank screen - CreateEvent

### DISCUSSION THREAD

**Discussion by Oupsla** [ionic-team/capacitor#4366] [Q&A] [answered]
Hello folks :wave:

Trying to upgrade to capacitor 3.0 but encountered an error.
Manage to compile and working great on the web but got a blank screen on both iOs and Android. When inspecting the webview with DevTools, I can see an error : Capacitor.createEvent is not a function in fireWindowEvent

We maybe suspect it comes from capacitor-cordova-plugin but nothing sure (and we didn’t manage to remove the generation of those ones, if someone know how do to it ?)

From Android Studio console :

```
I/Capacitor/Console: File: http://192.168.1.13:3000/ - Line 331 - Msg: Error in Success callbackId: NetworkStatus1649682775 : TypeError: Capacitor.createEvent is not a function
```

From Chrome Dev Tools :

```
Uncaught TypeError: Capacitor.createEvent is not a function
    at Object.fireWindowEvent ((index):235)
    at Object.callbackFromNative ((index):332)
    at <anonymous>:1:9
```

Cross-post in forum : https://forum.ionicframework.com/t/capacitor-3-blank-screen-createevent/206760

**Comment by jcesarmobile** [comment]
Did you do this?
https://capacitorjs.com/docs/v3/updating/3-0#importing-capacitorcore

while it says it's "recommended", it's sort of mandatory

**Reply by Oupsla** [reply]
Yes already done.

But I found the problem ! 
My error was to do `ReactDOM.render` in an async method (cause I had to init some libs before).
I have extracted this outside my async method and it is working great now 😄

**Comment by Oupsla** [comment, marked as answer]
For futur readers : 

This error was caused (in my project) cause `ReactDOM.render` was inside an async method. Putting it outside of it has resolved this issue

**Comment by Dave-1** [comment]
any solution for angular-ionic apps?
